main.cpp

```diff
@@ -8,189 +8,52 @@
-
-    DIR *dir;
-    struct dirent *ent;
-    string entry_name;
-    if ((dir = opendir (SysfsOnewireDevicesPath.c_str())) != NULL) {
-        /* print all the files and directories within directory */
-        while ((ent = readdir (dir)) != NULL) {
-            printf ("%s\n", ent->d_name);
-            entry_name = ent->d_name;
-            if (StringStartsWith(entry_name, "28-") ||
-                StringStartsWith(entry_name, "10-") ||
-                StringStartsWith(entry_name, "22-") )
-            {
-                    current_channels.emplace_back(entry_name);
-            }
-        }
-        closedir (dir);
-    } else {
-        cerr << "ERROR: could not open directory " << SysfsOnewireDevicesPath << endl;
-    }
-
-    vector<TSysfsOnewireDevice> new_channels;
-    UnorderedVectorDifference(current_channels, Channels, new_channels);
-
-    vector<TSysfsOnewireDevice> absent_channels;
-    UnorderedVectorDifference(Channels, current_channels, absent_channels);
-
-
-    Channels.swap(current_channels);
-
-    for (const TSysfsOnewireDevice& device: new_channels) {
-        Publish(NULL, GetChannelTopic(device) + "/meta/type", "temperature", 0, true);
-    }
-
-    //delete retained messages for absent channels
-    for (const TSysfsOnewireDevice& device: absent_channels) {
-        Publish(NULL, GetChannelTopic(device) + "/meta/type", "", 0, true);
-        Publish(NULL, GetChannelTopic(device), "", 0, true);
-    }
-}
-
-void TMQTTOnewireHandler::OnMessage(const struct mosquitto_message *message)
-{
-    string topic = message->topic;
-    string controls_prefix = string("/devices/") + MQTTConfig.Id + "/controls/";
-    if (topic == Retained_old) {// if we get old_message it means that we've read all retained messages
-        Publish(NULL, Retained_old, "", 0, true);
-        unsubscribe(NULL, Retained_old.c_str());
-        unsubscribe(NULL, (controls_prefix + "+").c_str());
-        PrepareInit = false;
-    }else {
-        string device = topic.substr(controls_prefix.length(), topic.length());
-        for (auto& current : Channels)
-            if (device == current.GetDeviceId())
-                return;
-        Channels.emplace_back(device);
-    }
-
-}
-
-void TMQTTOnewireHandler::OnSubscribe(int mid, int qos_count, const int *granted_qos)
-{
-	printf("Subscription succeeded.\n");
-}
-
-string TMQTTOnewireHandler::GetChannelTopic(const TSysfsOnewireDevice& device) {
-    static string controls_prefix = string("/devices/") + MQTTConfig.Id + "/controls/";
-    return (controls_prefix + device.GetDeviceId());
-}
-
-void TMQTTOnewireHandler::UpdateChannelValues() {
-
-    for (const TSysfsOnewireDevice& device: Channels) {
-        auto result = device.ReadTemperature();
-        if (result.Defined()) {
-            Publish(NULL, GetChannelTopic(device), StringFormat("%g",*result), 0, true); // Publish current value (make retained)
-        }
-
-    }
-}
 
+const auto WBMQTT_DB_FILE = "/var/lib/wb-homa-w1/libwbmqtt.db";
```

Comment:
Make sure that you create this folder (/var/lib/wb-homa-w1) at install, otherwise driver will fail at startup on fresh install.